### PR TITLE
dts: i2c: cleanup CONFIG_I2C_x_IRQ_PRI

### PR DESCRIPTION
**Pull request by galak**
The majority of cases of CONFIG_I2C_x_IRQ_PRI should be
DT_I2C_x_IRQ_PRI.  So go ahead and fix them up.  Only the i2c_nios
driver still uses Kconfig for getting priority.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -53,7 +53,6 @@ zephyr_include_directories(
   ${SOC_DIR}/${ARCH}/${SOC_PATH}
   ${SOC_DIR}/${ARCH}/${SOC_PATH}/include
   ${SOC_DIR}/${ARCH}/${SOC_FAMILY}/include
-  ${BOARD_DIR}
```
Comment (SebastianBoe) [inline CMakeLists.txt]: The commit message should explain why this is removed.

(And probably, this should be in it's own commit)

Comment (galak) [reply CMakeLists.txt]: oops, that snuck in, wasn't meant to be part of this PR.

**Review by ioannisg — APPROVED** [review]
nRF52 boards' fixups look ok to me.

**Review by ioannisg — CHANGES_REQUESTED** [review]
nRFx i2c driver is not updated, though.

**Comment by galak** [comment]
> nRFx i2c driver is not updated, though.

I think that is fixed now.

**Review by ioannisg — APPROVED** [review]

**Review by nashif — APPROVED** [review]
